subsys/net/lib/sockets/Kconfig

```diff
@@ -43,6 +43,19 @@ config NET_SOCKETS_ENABLE_DTLS
 	help
 	  Enable DTLS socket support. By default only TLS over TCP is supported.
 
+config NET_SOCKETS_DTLS_TIMEOUT
```

Comment:
TLS (and TCP) has the same problem in general. We need to offer more general solutions, e.g. implement SO_RCVTIMEO/SO_SNDTIMEO sockopts, while still allow to set specific default (instead of "no timeout") via Kconfig.

Reply:
Does TLS have the same problem? What I wanted to protect against with this timeout is TLS resources lockup in case peer went down. TLS is protected by TCP in this case - the connection will time-out at TCP level and it'll be closed. UDP is missing such a feature, hence extra effort is needed to prevent TLS context lockup.

As per `mbedTLS`, read timeout is DTLS specific.

Reply:
> TLS is protected by TCP in this case - the connection will time-out at TCP level and it'll be closed.

Classical TCP doesn't have any timeouts, it was designed with hardcore reliability in mind. You could detect some "timeout" on send, but there's even not possible to detect timeout on recv in classical TCP. An example I like is that there can be recv() call waiting for a year. Then someone cuts the wires. They are broken for next 10 years. Then someone fixes them. recv() will be able to receive data from the other connected peer as if nothing happened.

There're *options* and *extensions* to TCP to address that. They're all particular, and not even (fully) implemented in Zephyr. And in general, only a specific application may know they best way to deal with timeouts.

> As per mbedTLS, read timeout is DTLS specific.

So, what we're doing here is taking specific protocols (TCP, UDP, TLS, DTLS), their specific implementations (Zephyr native IP stack, mbedTLS, etc.), and trying to expose them via the standardized POSIX Sockets API. My call is exactly about that: we should strive for all the following: a) provide adequate timeout support (as needed by most real-world apps); b) make it full coverage (not just DTLS for example); c) adjustable via standard Sockets API calls; d) still standards compliant (allowing for infinite timeout).

Nothing to do particularly here now of course, just a call for thinking and forthcoming changes along the lines above.